source/blog/2017-08-14-ember-2-15-released.md

```diff
@@ -0,0 +1,377 @@
+
+An example of this API would be to transition to another route from a
+component:
+
+```js
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+  router: Ember.inject.service(),
+
+  actions: {
+    save(model) {
+      model.save().then(() => {
+        if (this.isDestroyed) { return; }
+        this.get('router').transitionTo('index');
+      });
+    }
+  }
+
+});
+```
+
+#### `{{mount}}` helper `model` argument
+
+Ember 2.15 implements
+[RFC #225](https://github.com/status200/rfcs/blob/ember-engines-mount-params/text/0000-ember-engines-mount-params.md) with some minor tweaks. The `{{mount}}`
+helper, used to invoke an Ember engine, now accepts the named argument of
+`model` to provide the model of the engine's application controller.
+
+For example, in an application you might pass some values from a template:
+
+```hbs
+<div>
+ <h1>Application template!</h1>
+ {{mount 'admin' model=(hash
+    title='Secret Admin'
+    signInButton=(component 'sign-in-button')
+ )}}
+</div>
+```
+
+And in an engine access those values on the `model` property:
+
+```hbs
+{{! admin/app/templates/application.hbs }}
+<h2>Admin area: {{model.title}}</h2>
+<div>
+  Please sign in: {{model.signInButton}}
+</div>
+```
+
+#### Other changes in 2.15
+
+Other changes include:
+
+* Ember 2.15 blueprints will no longer generate names for initializers and
+  instance initializers.
+* The deprecated API `_lookupFactory` has been removed in this release. See
+  the [deprecation
+  guide](https://www.emberjs.com/deprecations/v2.x/#toc_migrating-from-_lookupfactory-to-factoryfor)
+  for details about moving away from this API.
+* The Glimmer-VM rendering engine has been updated in this release, matching
+  Ember's rendering engine to that of the Glimmer.js library at its EmberConf
+  release. Included are VM improvement such as the "stack VM", improved
+  assertion stripping in production builds, and a more complete Glimmer-VM
+  implementation of the `(component` helper.
+
+No new deprecations are included in Ember 2.15
+
+For more details on changes in Ember.js 2.15, please review the
+[Ember.js 2.15.0 release page](https://github.com/emberjs/ember.js/releases/tag/v2.15.0).
+
+### Upcoming Changes in Ember.js 2.16-beta
+
+Ember.js 2.16-beta continues to improve the framework with minor bug fixes.
+Additionally there is a major change in usage coming in 2.16:
+
+#### Ember.js Modules API
+
+In Ember.js 2.16 the recommended way to access framework code in Ember
+applications will be via the JavaScript modules API described in
+[RFC #176](https://github.com/emberjs/rfcs/blob/master/text/0176-javascript-module-api.md). For example this basic component definition in 2.15:
+
+```js
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+  session: Ember.inject.service(),
+  title: 'The Curious Case'
+});
+```
+
+Would in 2.16 conventions be written as:
+
+```js
+import { inject } from '@ember/service';
```

Comment:
There is some discussion about this [here](https://github.com/ember-cli/ember-modules-codemod/issues/34), but at the moment the codemod will translate `session: Ember.inject.service(),` into `import { inject as service } from '@ember/service';` which has to be used `session: service()`.